components/kyma-environment-broker/internal/process/upgrade_kyma/initialisation.go

```diff
@@ -166,14 +166,27 @@ func (s *InitialisationStep) initializeUpgradeRuntimeRequest(operation internal.
 	}
 }
 
-func (s *InitialisationStep) configureKymaVersion(operation *internal.UpgradeKymaOperation) error {
+func (s *InitialisationStep) configureKymaVersion(operation *internal.UpgradeKymaOperation, orchestration *internal.Orchestration) error {
 	if !operation.RuntimeVersion.IsEmpty() {
 		return nil
 	}
-	version, err := s.runtimeVerConfigurator.ForUpgrade(*operation)
-	if err != nil {
-		return errors.Wrap(err, "while getting runtime version for upgrade")
+
+	var (
+		err     error
+		version *internal.RuntimeVersionData
+	)
+
+	// get Kyma version from orchestration or runtime parameters
+	if orchestration.HasVersion() {
+		version = internal.NewRuntimeVersionFromOrchestrationMapping(orchestration.Parameters.Version)
+	} else {
```

Comment:
Please change this logic to 

- store supplied kyma version in each upgrade operation
-  `runtimeVerConfigurator.ForUpgrade`() uses kyma version stored in operation. Eliminate the use of orchestration during operation processing.